packages/plugin-dts/src/utils.ts

```diff
@@ -70,3 +71,26 @@ export async function processDtsFiles(
     }
   }
 }
+
+export function processSourceEntry(
+  entryConfig: NonNullable<RsbuildConfig['source']>['entry'],
+): DtsEntry {
+  if (
+    entryConfig &&
+    Object.values(entryConfig).every((val) => typeof val === 'string')
+  ) {
+    return {
+      name: Object.keys(entryConfig)[0] as string,
+      path: Object.values(entryConfig)[0] as string,
+    };
+  }
+
+  logger.warn(
+    '@microsoft/api-extractor only support single entry of Record<string, string> type to bundle DTS.',
+  );
+
+  return {
```

Comment:
Is dts will be emitted if return from here?

Reply:
The bundless DTS will generated and api-extractor will throw error
![image](https://github.com/user-attachments/assets/ecc5595a-8ad6-4a88-8ed7-50189d6c67f6)

Reply:
1. bundleless output is useless as user want the bundled, can we detect the invalid option earlier and do not call the bundleless emit.
2. the log above (in line 88) should turn to be an error.

Reply:
I do a judgement and log like this
![image](https://github.com/user-attachments/assets/7fbc9ecd-ba32-4da5-b0c6-76b0440f646e)

Reply:
Could put the warn right above the error?

Reply:
![image](https://github.com/user-attachments/assets/09a61dce-5fd6-4b52-892f-2fe27b8046d8)
is this OK?